gui/src/components/deviceManagement/ManageDeviceConfiguration.js

```diff
@@ -78,23 +95,100 @@ class ManageDeviceConfiguration extends React.Component {
     return configurationFields;
   }
 
-  toggleDeletionModal = () => {
-    console.log('hi')
-    this.setState({deleteIsOpen: !this.state.deleteIsOpen});
+
+  toggleDeletionConfirmationModal = () => {
+    this.setState({deleteConfirmIsOpen: !this.state.deleteConfirmIsOpen});
+  }
+
+  toggleModifyConfirmationModal = () => {
+    this.setState({saveConfirmIsOpen: !this.state.saveConfirmIsOpen});
   }
+  
+  modifyDevice = () => {
+    xhr.open('PUT', modifyDeviceConfigurationBaseUrl + "?networkName=" + this.state.originalNetworkName + "&deviceId=" + this.state.originalDeviceName);
+    xhr.setRequestHeader("Content-Type", "application/json");
+
+    xhr.onreadystatechange = () => {
+      if (xhr.readyState === XMLHttpRequest.DONE) { // Once the request is done
+        if (xhr.status === 200) {
+          this.setState({
+            saveConfirmIsOpen: false
+          });
+          this.setState({
+            saveSuccessIsOpen: !this.state.successIsOpen
+          });      
+        }
+        else {
+          console.log(xhr);
```

Comment:
would be good to do something more than just logging when there is a failure

Reply:
Will update once api calls in master are updated from xhr in master